### PR TITLE
DM-28417: Support AT observing run 2020/01

### PR DESCRIPTION
**Pull request by tribeiro**
Fix error with atcs.stop_tracking.
Make some improvement to ATCS open/close dome methods.

### CODE REVIEW

File: python/lsst/ts/observatory/control/auxtel/atcs.py [hunk 1]
```diff
@@ -553,25 +556,52 @@ async def shutdown(self):
...
+                (
+                    await self.rem.atdome.evt_mainDoorState.aget(
+                        timeout=self.fast_timeout
+                    )
+                ).state
+            )
+
+            if shutter_pos != ATDome.ShutterDoorState.OPENED:
+                self.log.debug(
+                    f"Open shutter command returned while shutter still {shutter_pos!r}."
```
Comment (patrickingraham) [inline python/lsst/ts/observatory/control/auxtel/atcs.py:589]: why does this happen? is it a bug that needs fixing in ATDome? should we be referencing a ticket number?

File: python/lsst/ts/observatory/control/auxtel/atcs.py [hunk 2]
```diff
@@ -594,26 +624,56 @@ async def home_dome(self):
...
-            ]
+            shutter_pos = ATDome.ShutterDoorState(
+                (
+                    await self.rem.atdome.evt_mainDoorState.next(
+                        flush=False, timeout=self.open_dome_shutter_time
+                    )
+                ).state
+            )
+
+            if shutter_pos != ATDome.ShutterDoorState.CLOSED:
```
Comment (patrickingraham) [inline python/lsst/ts/observatory/control/auxtel/atcs.py:658]: same comment

File: python/lsst/ts/observatory/control/auxtel/atcs.py [hunk 3]
```diff
@@ -908,9 +1007,16 @@ def flush_offset_events(self):
...
+        while True:
+            in_position = await self.rem.atmcs.evt_allAxesInPosition.next(
+                flush=False, timeout=self.tel_settle_time
+            )
+
+            if in_position.inPosition:
+                self.log.debug("All axes in position.")
+                return in_position.inPosition
+            else:
+                self.log.debug("Telescope not in position.")
```
Comment (patrickingraham) [inline python/lsst/ts/observatory/control/auxtel/atcs.py:1019]: should this not raise an error?

**Review by patrickingraham — CHANGES_REQUESTED** [review]
code itself is fine, but i'm missing the context as to why it needs to exist...
Maybe just needs a comment or two explaining why?

File: python/lsst/ts/observatory/control/auxtel/atcs.py (hunk 3 above)
Comment (tribeiro) [reply python/lsst/ts/observatory/control/auxtel/atcs.py:1019]: No, if `in_position.inPosition = False` it means the telescope is not in position *yet*, so then you proceed to wait for the event again, until it is `True`. Also, note that, when we send an offset command the mount will first issue an `in_position.inPosition=False` (when the offset start) then `in_position.inPosition=True` (when the offset is complete).

File: python/lsst/ts/observatory/control/auxtel/atcs.py (hunk 1 above)
Comment (tribeiro) [reply python/lsst/ts/observatory/control/auxtel/atcs.py:589]: This (and the next one) is to handle the bug in ATDome. Yes, this should be removed once ATDome is fixed, *but* at the same time, this is not the main code base. It is a patch for "release/0.4", to work with the observatory during the run.

Comment (patrickingraham) [reply python/lsst/ts/observatory/control/auxtel/atcs.py:589]: ah i missed it was the patch. Sorry

**Review by patrickingraham — APPROVED** [review]
Comments addressed. Approved

**Pull request merged**